packages/markers/src/browser/problem/problem-composite-tree-node.spec.ts

```diff
@@ -0,0 +1,310 @@
+                expect(markerInfoNodeB.previousSibling).to.equal(markerInfoNodeA);
+            });
+            it('should sort errors higher than infos', () => {
+                const markerA = createMockMarker({ start: { line: 0, character: 10 }, end: { line: 0, character: 10 } }, DiagnosticSeverity.Information);
+                const markerB = createMockMarker({ start: { line: 0, character: 10 }, end: { line: 0, character: 10 } }, DiagnosticSeverity.Error);
+                const nodeA = createMockMarkerNode(markerA);
+                const nodeB = createMockMarkerNode(markerB);
+                const markerInfoNodeA = createMarkerInfo('1', new URI('a'), [nodeA]);
+                const markerInfoNodeB = createMarkerInfo('2', new URI('b'), [nodeB]);
+                ProblemCompositeTreeNode.addChild(rootNode, markerInfoNodeA, [markerA]);
+                ProblemCompositeTreeNode.addChild(rootNode, markerInfoNodeB, [markerB]);
+
+                expect(rootNode.children.length).to.equal(2);
+                expect(rootNode.children[0]).to.equal(markerInfoNodeB);
+                expect(markerInfoNodeB.nextSibling).to.equal(markerInfoNodeA);
+                expect(rootNode.children[1]).to.equal(markerInfoNodeA);
+                expect(markerInfoNodeA.previousSibling).to.equal(markerInfoNodeB);
+            });
+            it('should sort errors higher than hints', () => {
+                const markerA = createMockMarker({ start: { line: 0, character: 10 }, end: { line: 0, character: 10 } }, DiagnosticSeverity.Error);
+                const markerB = createMockMarker({ start: { line: 0, character: 10 }, end: { line: 0, character: 10 } }, DiagnosticSeverity.Hint);
+                const nodeA = createMockMarkerNode(markerA);
+                const nodeB = createMockMarkerNode(markerB);
+                const markerInfoNodeA = createMarkerInfo('1', new URI('a'), [nodeA]);
+                const markerInfoNodeB = createMarkerInfo('2', new URI('b'), [nodeB]);
+                ProblemCompositeTreeNode.addChild(rootNode, markerInfoNodeA, [markerA]);
+                ProblemCompositeTreeNode.addChild(rootNode, markerInfoNodeB, [markerB]);
+
+                expect(rootNode.children.length).to.equal(2);
+                expect(rootNode.children[0]).to.equal(markerInfoNodeA);
+                expect(markerInfoNodeA.nextSibling).to.equal(markerInfoNodeB);
+                expect(rootNode.children[1]).to.equal(markerInfoNodeB);
+                expect(markerInfoNodeB.previousSibling).to.equal(markerInfoNodeA);
+            });
+            it('should sort warnings higher than infos', () => {
+                const markerA = createMockMarker({ start: { line: 0, character: 10 }, end: { line: 0, character: 10 } }, DiagnosticSeverity.Error);
+                const markerB = createMockMarker({ start: { line: 0, character: 10 }, end: { line: 0, character: 10 } }, DiagnosticSeverity.Hint);
```

Comment:
DiagnosticSeverity should be Warning and Info

Reply:
@LukasBoll any reason not to change the `severity` here? The test-case is for `should sort warnings higher than infos` but we are comparing to `hint`.

Reply:
No, I have overlooked that one. I will fix it.

Reply:
Ok, the test should be correct now.